addons/mrp/models/mrp_workcenter.py

```diff
@@ -195,12 +259,45 @@ def unblock(self):
     def create(self, vals_list):
         # resource_type is 'human' by default. As we are not living in
         # /r/latestagecapitalism, workcenters are 'material'
-        records = super(MrpWorkcenter, self.with_context(default_resource_type='material')).create(vals_list)
-        return records
+
+        resources_vals_list = []
+        calendar_ids = [vals['resource_calendar_id'] for vals in vals_list if vals.get('resource_calendar_id')]
+        calendars_tz = {calendar.id: calendar.tz for calendar in self.env['resource.calendar'].browse(calendar_ids)}
+        for vals in vals_list:
+            if not vals.get('resource_ids'):
+                workorder_simultaneous = vals.get('simultaneous_workorder') if vals.get('simultaneous_workorder') else 1
+                resources = []
+                for _ in range(workorder_simultaneous):
+                    resources.append(self._prepare_resource_values(
+                        vals,
+                        vals.pop('tz', False) or calendars_tz.get(vals.get('resource_calendar_id'))
+                    ))
+                resources_vals_list.append(resources)
+        if resources_vals_list:
+            resources = []
+            for r in resources_vals_list:
+                resources.append(self.env['resource.resource'].create(r))
```

Comment:
- `_` is already used for the translation;
- `resources` is not needed, we can append the create values directly in `resources_vals_list`;
- By doing that, no need to loop to create the resource
```suggestion
                for _i in range(workorder_simultaneous):
                    resources_vals_list.append(self._prepare_resource_values(
                        vals,
                        vals.pop('tz', False) or calendars_tz.get(vals.get('resource_calendar_id'))
                    ))
        if resources_vals_list:
            resources = self.env['resource.resource'].create(resources_vals_list)
```